script.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Check if `git` exists in the current env.
+if ! command -v git &> /dev/null; then
+    echo "git could not be found"
+    exit 2
+fi
+
+ERRORS=`git log --grep='^\w+(\((\w|-)+\))?!?: .+$' --invert-grep -E --pretty=format:%s | awk 'END {print NR}'`
+echo "$ERRORS"
+
+if [[ -n $GITHUB_ACTIONS && $GITHUB_ACTIONS == true ]]; then
+    # Print to github_output only in a Github Actions environment.
+    echo "errors=$ERRORS" >> "$GITHUB_OUTPUT"
+fi
```

Comment:
The whole script can be put in the action `run` block, there is no need of a new file. If you fear half-execution, put everything inside a function and call the function. Also, inside the function you can run `set -euo pipefail` to stop the script at the first error.

Reply:
Using a new file for the script allow us to run it in other env and calling it in other ways. If you dislike it, I'll make the change.

Reply:
What other env do you have in mind?

Reply:
Local git hooks first of all, they run a script.

Reply:
The repo is your, I'll go with the action run ;)